### PR TITLE
[COM-97] return list in findByProductName

### PR DESCRIPTION
**Pull request by valentyna-koshelnyk**
https://1sresr3ada.atlassian.net/jira/software/c/projects/COM/boards/5?issueType=10009%2C10007%2C10008&quickFilter=4&selectedIssue=COM-97

https://1sresr3ada.atlassian.net/browse/COM-98

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,7 +6,8 @@ target/
 !**/src/test/**/target/
 .properties
 .yml
-
+ecom.log
+.gz
```
Comment (faruka-startsteps) [inline .gitignore:10]: I think this should be *.gz

File: src/main/java/com/startsteps/ecommerceapi/payload/response/ApiResponse.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.startsteps.ecommerceapi.payload.response;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class ApiResponse {
```
Comment (faruka-startsteps) [inline src/main/java/com/startsteps/ecommerceapi/payload/response/ApiResponse.java:10]: Maybe rename this to `MessageResponse` it is more clear what it represents

File: src/main/java/com/startsteps/ecommerceapi/utils/Default.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.startsteps.ecommerceapi.utils;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.CONSTRUCTOR)
+@Retention(RetentionPolicy.CLASS)
+public @interface Default {
```
Comment (faruka-startsteps) [inline src/main/java/com/startsteps/ecommerceapi/utils/Default.java:10]: is this needed?

**Review by faruka-startsteps — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (valentyna-koshelnyk) [reply .gitignore:10]: good one, thx! on it :)

File: src/main/java/com/startsteps/ecommerceapi/utils/Default.java (hunk 3 above)
Comment (valentyna-koshelnyk) [reply src/main/java/com/startsteps/ecommerceapi/utils/Default.java:10]: without custom default annotation mapstruct has conflicts with multiple constructors :(

File: src/main/java/com/startsteps/ecommerceapi/payload/response/ApiResponse.java (hunk 2 above)
Comment (valentyna-koshelnyk) [reply src/main/java/com/startsteps/ecommerceapi/payload/response/ApiResponse.java:10]: the thing is that I have my message response reserved for the authentication and it has field jwt. would I need to return jwt for the rest or is it better to keep a single responsibility class ApiResponse?

Comment (faruka-startsteps) [reply src/main/java/com/startsteps/ecommerceapi/payload/response/ApiResponse.java:10]: it's better to `JwtResponse` and `MessageResponse`. Both are more clear that way

Comment (valentyna-koshelnyk) [reply src/main/java/com/startsteps/ecommerceapi/payload/response/ApiResponse.java:10]: I see, thx!

**Pull request merged**